design/decisions/why-lint-package/Why-flake8.md

```diff
@@ -0,0 +1,123 @@
+capabilities, making them suitable for different project needs. The
+primary alternatives evaluated were Pylint, Black, and ruff.
+
+### Pylint
+
+- **Features**: Pylint is known for its comprehensive code analysis
+  capabilities. It is particularly adept at identifying potential bugs,
+  adhering to coding standards, and highlighting design issues in the
+  codebase.
+- **Configurability**: One of the standout features of Pylint is its
+  high configurability. It allows developers to adjust the strictness of
+  linting rules, making it highly adaptable to specific project
+  preferences and coding standards.
+- **Feedback**: Pylint provides extensive feedback on code quality,
+  including scores, which can be instrumental in maintaining high coding
+  standards across a project.
+
+### Black
+
+- **Focus**: Black is primarily a code formatting tool. It automatically
+  reformats Python code to adhere to a consistent style, reducing the
+  cognitive load associated with manual styling.
+- **Consistency**: With its opinionated approach to code styling, Black
+  ensures a uniform code style across the project. This consistency is
+  particularly beneficial in projects where readability and
+  collaboration are prioritized.
+- **Integration**: Black easily integrates into development workflows,
+  often used as a pre-commit hook or part of the continuous integration
+  process, ensuring code style consistency with minimal developer
+  intervention.
+
+### Ruff
+
+- **Speed**: Ruff is a relatively new linting tool known for its speed,
+  leveraging Rust's performance to deliver quick linting results.
+- **Simplicity**: It offers a simpler setup and operation, appealing for
+  projects looking for straightforward linting solutions.
+- **Emerging Tool**: While promising, ruff is still gaining traction and
+  may not yet have the extensive community support and plugin ecosystem
+  compared to more established tools.
+
+## Decision Outcome
+
+### Comparison with Flake8
+
+- **Versatility**: While Pylint offers detailed analysis, Black excels
+  in formatting, and ruff impresses with its speed, Flake8 strikes a
+  balance between enforcing style guidelines, maintaining code quality,
+  and providing a wide range of features through its integration of
+  tools like pyflakes, pycodestyle, and mccabe.
+
+- **Customization**: Flake8’s modularity allows for significant
+  customization, enabling developers to adapt the linting process to
+  their specific project needs without sacrificing essential linting
+  features.
+- **Community and Ecosystem**: The widespread adoption of Flake8 in the
+  Python community has led to a robust ecosystem of plugins and active
+  community support. This ecosystem ensures continual improvement and
+  responsiveness to emerging needs in Python development.
+
+### Conclusion
+
+In summary, while Pylint, Black, and ruff each have their strengths in
+specific areas, Flake8 was chosen for its balanced approach,
```

Comment:
I'm not sure it's very clear why we should use flake8 rather than Ruff (especially since it is designed to replace these other tools and still be compatible with them).